### PR TITLE
Use CAPTURE_CMAKE_ERROR instead of RETURN_VALUE or as well as NUMBER_ERRORS

### PR DESCRIPTION
**Pull request by jamiesnape**
Apparently the original plan to reuse RETURN_VALUE changed to adding CAPTURE_CMAKE_ERROR instead.

### CODE REVIEW

**Comment by jamiesnape** [comment]
PTAL +@mwoehlke-kitware.

**Comment by jamiesnape** [comment]
https://drake-jenkins.csail.mit.edu/job/linux-clang-ninja-experimental/47/
https://drake-jenkins.csail.mit.edu/job/mac-clang-ninja-experimental/218/
https://drake-jenkins.csail.mit.edu/job/mac-clang-xcode-experimental/7/

File: driver/configurations/generic/step-drake.cmake [hunk 1]
```diff
@@ -98,9 +99,9 @@ endif()
 
 if(DASHBOARD_INSTALL)
   ctest_submit(RETRY_COUNT 4 RETRY_DELAY 15
```
Comment (mwoehlke-kitware) [inline driver/configurations/generic/step-drake.cmake]: BTW, I wonder if this `ctest_submit` is needed?

File: driver/configurations/generic/step-post-drake.cmake [hunk 2]
```diff
@@ -18,8 +18,9 @@ endif()
 
 # Run the build
 ctest_build(BUILD "${CTEST_BINARY_DIRECTORY}" APPEND
-  NUMBER_ERRORS DASHBOARD_SUPERBUILD_NUMBER_BUILD_ERRORS QUIET)
-if(DASHBOARD_SUPERBUILD_NUMBER_BUILD_ERRORS GREATER 0)
+  NUMBER_ERRORS DASHBOARD_SUPERBUILD_NUMBER_BUILD_ERRORS
+  CAPTURE_CMAKE_ERROR DASHBOARD_BUILD_RETURN_VALUE QUIET)
```
Comment (mwoehlke-kitware) [inline driver/configurations/generic/step-post-drake.cmake]: pedantic: `DASHBOARD_SUPERBUILD_BUILD_RETURN_VALUE`?

**Comment by jamiesnape** [comment]
https://drake-jenkins.csail.mit.edu/job/linux-clang-ninja-experimental/51/
https://drake-jenkins.csail.mit.edu/job/mac-clang-ninja-experimental/220/
https://drake-jenkins.csail.mit.edu/job/mac-clang-xcode-experimental/8/

**Pull request merged**